### PR TITLE
Fix multimodal and moe issue

### PR DESCRIPTION
**Pull request by WeiweiZhang1**
fix mixtral-8x7b model quantization issue

### CODE REVIEW

File: auto_round/export/export_to_autoround/export.py [hunk 1]
```diff
@@ -166,6 +166,7 @@ def save_quantized_as_autoround(output_dir, inplace=True, backend="auto_round:ex
 
     model = kwargs["model"]
     multimodal = kwargs["multimodal"]
+    safe_serialization = True if 'safe_serialization' not in kwargs.keys() else  kwargs["safe_serialization"]
```
Comment (wenhuach21) [inline auto_round/export/export_to_autoround/export.py:169]: why need this

File: auto_round/special_model_handler.py [hunk 2]
```diff
@@ -12,11 +12,13 @@
...
 
-from .utils import logger, torch
+import torch
 
 share_attention_mask_tuple = ("baichuan",)
 special_states_dim_tuple = ("chatglm",)
-not_share_position_ids_tuple = ("llava","phi3_v")
+not_share_position_ids_tuple = ("llava", "phi3_v",)
+Vison_blocks_tuple = ("vision", "visual",)
+MoE_blocks_tuple = ("moe", "expert",)
```
Comment (wenhuach21) [inline auto_round/special_model_handler.py]: do not handle moe as special model . there are many moe models, which may not name block as expert.

File: auto_round/export/export_to_autoround/export.py (hunk 1 above)
Comment (WeiweiZhang1) [reply auto_round/export/export_to_autoround/export.py:169]: for phi3-vision save model process

File: auto_round/autoround.py [hunk 3]
```diff
@@ -140,11 +141,12 @@ def __init__(
             dynamic_max_gap: int = -1,
             data_type: str = "int",
             scale_dtype: str = "fp16",
-            multimodal:bool = False,
             act_bits: int = 32,
             act_group_size: int = None,
             act_sym: bool = None,
             act_dynamic: bool = True,
+            multimodal:bool = False,
+            quant_vision:bool = False,
```
Comment (wenhuach21) [inline auto_round/autoround.py]: how about just pass the block_names [list of list] to support this, while the example of mutlimodel could support this args

File: auto_round/export/export_to_autoround/export.py (hunk 1 above)
Comment (wenhuach21) [reply auto_round/export/export_to_autoround/export.py:169]: if safe serialization  has no effect for other models, we could use it as the default in the future

File: auto_round/utils.py [hunk 4]
```diff
@@ -232,22 +232,75 @@ def check_is_cpu(device):
     return device == torch.device("cpu") or device == "cpu"
 
 
-def get_block_names(model, multimodal=False):
+def validate_modules(module_names):
+        """
+        Validate a list of modules validity.
```
Comment (wenhuach21) [inline auto_round/utils.py]: could not understand the comment

File: auto_round/autoround.py [hunk 5]
```diff
@@ -100,12 +99,11 @@ class AutoRound(object):
         data_type (str): The data type to be used (default is "int").
         scale_dtype (str): The data type of quantization scale to be used (default is "float16"), different kernels
                            have different choices.
-        multimodal(bool): Enable multimodal model quantization, (default is "False").
         act_bits (int): Number of bits for activation quantization. Default is 32.
         act_group_size (int): Group size for activation quantization. Default is None.
         act_sym (bool): Whether to use symmetric activation quantization. Default is None.
         act_dynamic (bool): Whether to use dynamic activation quantization. Default is True.
-
+        quant_block_list (bool): A list whose elements are list of block's layer names to be quantized.
```
Comment (wenhuach21) [inline auto_round/autoround.py]: not bool

File: auto_round/utils.py [hunk 6]
```diff
@@ -234,13 +234,13 @@ def check_is_cpu(device):
 
 def validate_modules(module_names):
         """
-        Validate a list of modules validity.
```
Comment (wenhuach21) [inline auto_round/utils.py]: where it will be used

**Review by wenhuach21 — APPROVED** [review]

File: auto_round/utils.py (hunk 6 above)
Comment (WeiweiZhang1) [reply auto_round/utils.py]: in get_multimodal_block_names

**Pull request merged**